### ISSUE TITLE
Load model with keras

### ISSUE THREAD

**Issue by AzazelHD** [shreyas-bk/U-2-Net-Demo#9]
Hi.

Is there anyway i can load this model using keras.models.load_model?

Pls i will really appreciate it

**Comment by shreyas-bk** [comment]
Hi, there are implementations like [this](https://github.com/Akhilesh64/Image-Segmentation-U-2-Net/blob/main/predict.py) that may help.

**Comment by AzazelHD** [comment]
Mmmm I dont think so. The predicted mask is way different from the original mask. I just want to load U2NET.pth weights into keras model. The original weights

**Comment by shreyas-bk** [comment]
I have been able to convert to a TensorFlow .pb saved model (using ONNX) if that helps, but currently I am unable to convert to Keras h5. I will try again when I get more time.

**Comment by AzazelHD** [comment]
Yeah same. From pth to onnx can be done. But I havent found a way to make it .h5.
Also, if you can train the model with same datasets on keras and share a link with keras model, I will appreciate a lot.

Btw, thanks for helping me with this

**Comment by shreyas-bk** [comment]
No problem
I was able to get a Keras h5 file using pytorch2keras, however I had to change the upsampling mode from bilinear to nearest. Nevertheless, the predicted masks seem the same. Let me know if you want me to share it.

**Comment by AzazelHD** [comment]
Yeah! I got that error complaining about upsampling layers. And sure, if you could share you will make me  big favour ^^

Thank you very much dude

**Comment by shreyas-bk** [comment]
I made this repository : [U-2-Net-Keras](https://github.com/shreyas-bk/U-2-Net-Keras). You can find both the Keras h5 saved model as well as the notebook I used to make it. With the code in the inference cell of the notebook, you should get the right results.
Let me know if it works, and please star if it helped :)
I will also upload the TensorFlow version if anyone requires it.

**Comment by AzazelHD** [comment]
Omg sorry I am derping. Im new here. How can I give u stars?

**Comment by shreyas-bk** [comment]
There are buttons at the top of the page, (watch, star and fork), you can just click star
BTW if there are any further issues with the keras version let me know

**Comment by AzazelHD** [comment]
I mean, probably your code works perfectly. I think i am just idiot. Could I add you to discord by chance?
BTW Star given ^^ thanks for your time

**Comment by AzazelHD** [comment]
Oh I just noticed... You uploaded U2NET**P**... Could I ask for U2NET?

**Comment by shreyas-bk** [comment]
For U2NET, you can just download the original weight file to the [colab](https://colab.research.google.com/github/shreyas-bk/U-2-Net-Keras/blob/main/U_2_Netp_Demonstration_Colab_PyTorch2Keras.ipynb?authuser=1) in the repo, then just change u2netp to u2net in the setup cell. Either way, I'll also upload the Keras h5 for it, probably tomorrow. 
For discord related things let's first connect by email (shreykera7@gmail.com) so that I can close this issue?
